src/Network/Request.php

```diff
@@ -1375,6 +1381,42 @@ public function setInput($input)
     }
 
     /**
+     * Update the request with a new request data element.
+     *
+     * Returns an updated request object. This method returns
+     * a *new* request object and does not mutate the request in-place.
+     *
+     * @param string $name The dot separated path to insert $value at.
+     * @param mixed $value The value to insert into the request data.
+     * @return self
```

Comment:
If people would stick with what phpDoc says, then `$this` wouldn't be a recognized type accoring to https://www.phpdoc.org/docs/latest/references/phpdoc/types.html. It is however listed in https://phpdoc.org/docs/latest/guides/types.html#keywords, together with `self` and `static`.

Most Major IDEs support `$this`, `self` and `static`, and given how they interpret things (that is `$this` is the current object, `self` an instance of the current class _where the method is defined_, and `static` is an instance of _the last inheritor_ of the current class), I'd say that neither `$this` nor `self` is correct, but `static` unless the class would be marked as `final`.

As @markstory said, it's not `$this` anymore, but an instance of the current class, ie a _different_ object, and inheritance could possibly be involed. While `$this` or `static` would make no difference for IDEs currently (at least I personally know none that would), that may change when they become smarter. Code analysis tools surely could make use of this difference.